src/main/java/spoon/support/reflect/cu/SourcePositionImpl.java

```diff
@@ -62,53 +62,53 @@ private int searchLineNumber(int[] startLineIndexes, int position) {
 	/**
 	 * Search the column number
 	 */
-	private int searchColumnNumber(int[] startLineIndexes, int position) {
-		if (startLineIndexes == null) {
+	private int searchColumnNumber(int position) {
+		if (lineSeparatorPositions == null) {
 			return 1;
 		}
-		int length = startLineIndexes.length;
+		int length = lineSeparatorPositions.length;
 		if (length == 0) {
 			return 1;
 		}
 		int i = 0;
-		for (i = 0; i < startLineIndexes.length - 1; i++) {
-			if (startLineIndexes[i] < position && (startLineIndexes[i + 1] > position)) {
-				return position - startLineIndexes[i];
+		for (i = 0; i < lineSeparatorPositions.length - 1; i++) {
+			if (lineSeparatorPositions[i] < position && (lineSeparatorPositions[i + 1] > position)) {
+				return position - lineSeparatorPositions[i];
 			}
 		}
 		int tabCount = 0;
 		int tabSize = 0;
 		if (getCompilationUnit() != null) {
 			tabSize = getCompilationUnit().getFactory().getEnvironment().getTabulationSize();
 			String source = getCompilationUnit().getOriginalSourceCode();
-			for (int j = startLineIndexes[i]; j < position; j++) {
+			for (int j = lineSeparatorPositions[i]; j < position; j++) {
 				if (source.charAt(j) == '\t') {
 					tabCount++;
 				}
 			}
 		}
-		return (position - startLineIndexes[i]) - tabCount + (tabCount * tabSize);
+		return (position - lineSeparatorPositions[i]) - tabCount + (tabCount * tabSize);
 	}
 
 	int[] lineSeparatorPositions;
 
-	private int sourceStart, sourceEnd, line;
+	private int sourceStartName, sourceEnd, sourceStart, line = -1;
 
 	public SourcePositionImpl(CompilationUnit compilationUnit, int sourceStartDeclaration, int sourceStartSource, int sourceEnd, int[] lineSeparatorPositions) {
```

Comment:
A bit of Javadocumenting seems welcomed here since the name of the parameters are not really self-intuitive. What is sourceStartDeclaration? What is the difference with sourceStartSource? And why do we need lineSeparatorPositions?